contracts/examples/WitnetRNG.sol

```diff
@@ -0,0 +1,234 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.7.0 <0.9.0;
+pragma experimental ABIEncoderV2;
+
+import "../UsingWitnet.sol";
+import "../requests/WitnetRequestRandomness.sol";
+
+/// @title WitnetRNG: EVM-agnostic trustless random number generator, based on the Witnet oracle. 
+/// @author The Witnet Foundation.
+contract WitnetRNG
+    is
+        UsingWitnet,
+        WitnetRequestRandomness
+{
+    uint256 internal nonce;
+
+    /// Include an address to specify the immutable WitnetRequestBoard entrypoint address.
+    /// @param _wrb The WitnetRequestBoard immutable entrypoint address.
+    constructor(WitnetRequestBoard _wrb)
+        UsingWitnet(_wrb)
+    {}
+
+    /// @dev Modifier: makes sure latest request was already solved.
+    modifier notPending {
+        require(isReady(), "WitnetRNG: pending randomize");
+        _;
+    }
+
+    /// Gets randomness generated upon latest request.
+    /// @dev Returns 0x0...0 if not yet solved, or 0xf...f if randomness could not get solved by Witnet for any 
+    /// @dev unexpected reason. Fails if `randomize()` was never called before.
+    function getRandomness()
```

Comment:
I think it would be a better API to have this function revert in case of error, because it is a public function so any user may call it and forget to check this special conditions.

Reply:
The issue would then be how to enable clients in general, and the contract owner in particular, a simple way to detect the weird case in which the WitnetRequestRandomness resolves with an error (i.e. 0xf...f). Perhaps turning @dev comment as part of the @notice description would help unaware developers.

With respect letting 0x0 as returned value when no randomize() was ever called, seems fair enough to me.